bluetooth/sql/functions/reader_status_history.sql

```diff
@@ -0,0 +1,122 @@
+	insert_value date)
+    RETURNS TABLE(an_id bigint, last_received date, status text, ddown integer, start_route_point_id text, start_detector text, end_route_point_id text, end_detector text)
+    LANGUAGE 'plpgsql'
+
+    COST 100
+    VOLATILE 
+AS $BODY$	
+	begin
+		--return query
+			with x AS 
+			(select DISTINCT(analysis_id), 
+			 CASE
+			 WHEN MAX (datetime_bin::date)>= (insert_value) then (insert_value)
+			 ELSE MAX (datetime_bin::date)
+			 END 
+			 as last_reported,
+			CASE
+			WHEN  MAX (datetime_bin::date)>= (insert_value) then 'True'::text
+		else 'False'::text
+		END 
+		AS route_status
+			 ,CASE
+		WHEN max(aggr_5min.datetime_bin::date) >= (insert_value::text::date) THEN insert_value::date
+                    ELSE (insert_value::date)
+		END AS dt
+		from 
+		bluetooth.all_analyses
+		LEFT JOIN bluetooth.aggr_5min USING (analysis_id) 
+		GROUP BY analysis_id)
+		, y AS (
+		 SELECT all_analyses.analysis_id,
+    (all_analyses.route_points -> 0) ->> 'id'::text AS start_route_point_id,
+    (all_analyses.route_points -> 0) ->> 'name'::text AS start_detector,
+    (all_analyses.route_points -> 1) ->> 'id'::text AS end_route_point_id,
+    (all_analyses.route_points -> 1) ->> 'name'::text AS end_detector
+   FROM bluetooth.all_analyses
+		)
+  , z AS (
+         SELECT x.analysis_id,
+            x.last_reported,
+            x.route_status,
+            x.dt,
+            y.start_route_point_id,
+            y.start_detector,
+            y.end_route_point_id,
+            y.end_detector
+           FROM x
+             LEFT JOIN y USING (analysis_id)
+        )
+		, b AS (
+         SELECT z.analysis_id,
+            z.start_detector AS detector_name,
+            z.route_status,
+            z.last_reported,
+			z.dt
+           FROM z
+        )
+		, c AS (
+         SELECT b.analysis_id,
+            b.detector_name,
+            b.route_status,
+            b.last_reported,
+			b.dt
+           FROM b
+        UNION
+         SELECT z.analysis_id,
+            z.end_detector AS detector_name,
+            z.route_status,
+            z.last_reported,
+			z.dt
+           FROM z
+        )
+		, active AS (
+         SELECT DISTINCT c.detector_name,
+            max(c.last_reported) AS max,
+            c.route_status,
+            bluetooth.detectors_history_final.reader_id as id,
+			c.dt
+           FROM c
+             LEFT JOIN bluetooth.detectors_history_final ON c.detector_name = bluetooth.detectors_history_final.read_name::text
+          WHERE c.route_status = 'True'::text
+          GROUP BY c.route_status, c.detector_name, bluetooth.detectors_history_final.reader_id, c.dt
+        ), final as (
+ SELECT DISTINCT c.detector_name,
+    max(c.last_reported) AS max,
+    c.route_status,
+    bluetooth.detectors_history_final.reader_id,
+			c.dt
+   FROM c
+     LEFT JOIN bluetooth.detectors_history_final ON c.detector_name = bluetooth.detectors_history_final.read_name::text
```

Comment:
not exactly sure what is detectors_history_final as it is not bluetooth schema and not documented

Reply:
Found 5 `detectors_history_*` tables in schema `mohan`, 
`detectors_history_final`
`detectors_history`
`detectors_history_late`
`detectors_history_latest`
`detectors_history_corrected`

Reply:
seems like the reason to join to this table is to get the corresponding `reader_id` which is needed in the output. This should also be an inner join